tests/prague/eip7692_eof_v1/eip7069_extcall/test_address_space_extension.py

```diff
@@ -0,0 +1,227 @@
+                        + Op.SSTORE(slot_target_returndata, Op.MLOAD(0))
+                        + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1 + len(call_suffix),
+                    )
+                ],
+            )
+            if ase_ready_opcode
+            else Op.SSTORE(
+                slot_target_call_status,
+                target_opcode(Op.GAS, Op.CALLDATALOAD(0), *call_suffix),
+            )
+            + Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE)
+            + Op.SSTORE(slot_target_returndata, Op.MLOAD(0))
+            + Op.STOP,
+            nonce=1,
+        ),
+    }
+    match target_account_type:
+        case "empty":
+            # add no account
+            pass
+        case "EOA":
+            pre[Address(stripped_address)] = Account(code="", balance=10**18, nonce=9)
+        case "LegacyContract":
+            pre[Address(stripped_address)] = Account(
+                code=Op.MSTORE(0, Op.ADDRESS) + Op.RETURN(0, 32),
+                balance=0,
+                nonce=0,
+            )
+        case "EOFContract":
+            pre[Address(stripped_address)] = Account(
+                code=Container(
+                    sections=[
+                        Section.Code(
+                            code=Op.MSTORE(0, Op.ADDRESS) + Op.RETURN(0, 32),
+                            code_outputs=NON_RETURNING_SECTION,
+                            max_stack_height=2,
+                        )
+                    ],
+                ),
+                balance=0,
+                nonce=0,
+            )
+
+    target_storage: dict[int, int | bytes | Address] = {}
+    match target_account_type:
+        case "empty" | "EOA":
+            target_storage[slot_target_call_status] = (
+                EXTCALL_SUCCESS if ase_ready_opcode else CALL_SUCCESS
+            )
+        case "LegacyContract" | "EOFContract":
+            match target_opcode:
+                case Op.CALL | Op.STATICCALL:
+                    target_storage[slot_target_call_status] = CALL_SUCCESS
+                    # CALL and STATICCALL call will call the stripped address
+                    target_storage[slot_target_returndata] = stripped_address
+                case Op.CALLCODE | Op.DELEGATECALL:
+                    target_storage[slot_target_call_status] = CALL_SUCCESS
+                    # CALLCODE and DELEGATECALL call will call the stripped address
+                    # but will change the sender to self
+                    target_storage[slot_target_returndata] = address_caller
+                case Op.EXTCALL | Op.EXTSTATICCALL:
+                    target_storage[slot_target_call_status] = EXTCALL_SUCCESS
+                    # EXTCALL and EXTSTATICCALL will fault if calling an ASE address
+                    target_storage[slot_target_returndata] = 0 if ase_address else stripped_address
```

Comment:
Same as previous comment:
```suggestion
                    target_storage[slot_target_returndata] = exceptional_abort_canary if ase_address else stripped_address
```